src/event-bridge/rule.ts

```diff
@@ -21,11 +21,15 @@ import type { Event } from "./types";
  *
  * event is every event sent to the bus to be filtered. This argument is optional.
  */
-export type RulePredicateFunction<E, O extends E = E> =
-  | ((event: E) => event is O)
-  | ((event: E) => boolean);
+export type RulePredicateFunction<Evnt, OutEvnt extends Evnt = Evnt> =
+  | ((event: Evnt) => event is OutEvnt)
+  | ((event: Evnt) => boolean);
 
-export interface IRule<T extends Event> {
+/**
+ * @typeParam - Evnt - The original event type from the {@link EventBus}.
+ * @typeParam - OutEvnt - The narrowed event type after the predicate is applied.
+ */
+export interface IRule<in Evnt extends Event, out OutEvnt extends Evnt> {
```

Comment:
Why do we need two type parameters?

Reply:
`Evnt` is the event in. The input to the predicate on the rule.

`OutEvnt` is the narrowed type as the input to `pipe` or `map` or `when`.

Essentially splitting and showing the two sides of the predicate.

Reply:
Why do you need `Evnt`?

Reply:
Its being used on `integration: IntegrationWithEventBus<Evnt, Props>,`

But that doesn't seem right... will look.